src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt

```diff
@@ -306,76 +330,67 @@ abstract class FixupSession(
       if (!controller.isEligibleForInlineEdit(editor)) {
         return@withListenersMuted logger.warn("Inline edit not eligible")
       }
-      // Mark all the edit locations so the markers will move as we edit the document,
-      // preserving the original positions of the edits.
-      val markers = edits.mapNotNull { createMarkerForEdit(it) }
-      val sortedEdits = edits.zip(markers).sortedByDescending { it.second.startOffset }
-      // Apply the edits in a write action.
+
       WriteCommandAction.runWriteCommandAction(project) {
-        for ((edit, marker) in sortedEdits) {
-          try {
-            when (edit.type) {
-              "replace",
-              "delete" -> {
-                val action = ReplaceUndoableAction(project, session = this, edit, marker)
-                this.performedActions.add(action)
-                action.apply()
+        val currentActions =
+            edits.mapNotNull { edit ->
+              try {
+                when (edit.type) {
+                  "replace",
+                  "delete" -> ReplaceUndoableAction(project, edit, document)
+                  "insert" -> InsertUndoableAction(project, edit, document)
+                  else -> {
+                    logger.warn("Unknown edit type: ${edit.type}")
+                    null
+                  }
+                }
+              } catch (e: RuntimeException) {
+                throw EditCreationException(edit, e)
               }
-              "insert" -> {
-                val action = InsertUndoableAction(project, session = this, edit, marker)
-                this.performedActions.add(action)
-                action.apply()
-              }
-              else -> logger.warn("Unknown edit type: ${edit.type}")
             }
+
+        currentActions.forEach { action ->
+          try {
+            action.apply()
           } catch (e: RuntimeException) {
-            throw EditException(edit, marker, e)
+            throw EditExecutionException(action, e)
           }
         }
+
+        performedActions += currentActions
       }
     }
   }
 
-  private fun createMarkerForEdit(edit: TextEdit): RangeMarker? {
-    val startOffset: Int
-    val endOffset: Int
-    when (edit.type) {
-      "replace",
-      "delete" -> {
-        val range = edit.range ?: return null
-        startOffset = document.getLineStartOffset(range.start.line) + range.start.character
-        endOffset = document.getLineStartOffset(range.end.line) + range.end.character
-      }
-      "insert" -> {
-        val position = edit.position ?: return null
-        startOffset = document.getLineStartOffset(position.line) + position.character
-        endOffset = startOffset
-      }
-      else -> return null
-    }
-    return createMarker(startOffset, endOffset)
+  private fun adjustToDocumentRange(r: Range): Range {
+    val start = if (r.start.line < 0) Position(line = 0, character = r.start.character) else r.start
```

Comment:
In what situations is a negative `r.start.line` valid?

Reply:
Good question, I added comment in the code.